pkg/cloud/services/compute/instance.go

```diff
@@ -505,6 +512,28 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 	}
 
 	record.Eventf(eventObject, "SuccessfulCreatePort", "Created port %s with id %s", port.Name, port.ID)
+	// https://pkg.go.dev/github.com/gophercloud/gophercloud@v0.16.0/openstack/networking/v2/extensions/portsecurity
+	if portOpts.DisablePortLevelSecurity == true {
+		var portWithPortSecurityExtensions struct {
+			ports.Port
+			portsecurity.PortSecurityExt
+		}
+		// Get current status
+		err = ports.Get(s.networkClient, port.ID).ExtractInto(&portWithPortSecurityExtensions)
+		if portWithPortSecurityExtensions.PortSecurityEnabled == true {
+			iFalse := false
+			portUpdateOpts := ports.UpdateOpts{}
+			updateOpts := portsecurity.PortUpdateOptsExt{
+				UpdateOptsBuilder:   portUpdateOpts,
+				PortSecurityEnabled: &iFalse,
+			}
+			err := ports.Update(s.networkClient, port.ID, updateOpts).ExtractInto(&portWithPortSecurityExtensions)
+			if err != nil || portWithPortSecurityExtensions.PortSecurityEnabled == true {
+				return nil, fmt.Errorf("Unable to disable port security at port level: %v", err)
+			}
+		}
+
+	} // end of disablng port security
```

Comment:
typo
```suggestion
	} // end of disabling port security
```

Reply:
will remove the comment, added for ease of coding